pkg/controller/secretsmanager/secret/setup.go

```diff
@@ -172,18 +258,13 @@ func (e *hooks) isUpToDate(cr *svcapitypes.Secret, resp *svcsdk.DescribeSecretOu
 	case len(s.SecretBinary) != 0:
 		return bytes.Equal(payload, s.SecretBinary), nil
 	}
-	return false, errors.New("neither SecretString nor SecretBinary field is filled in the returned object")
+	return false, errors.New(errNoAWSValue)
 }
 
-func (e *hooks) getPayload(ctx context.Context, cr *svcapitypes.Secret) ([]byte, error) {
-	var ref *svcapitypes.SecretReference
-	switch {
-	case cr.Spec.ForProvider.StringSecretRef != nil:
-		ref = cr.Spec.ForProvider.StringSecretRef
-	case cr.Spec.ForProvider.BinarySecretRef != nil:
-		ref = cr.Spec.ForProvider.BinarySecretRef
-	default:
-		return nil, errors.New("neither binarySecretRef nor stringSecretRef is given")
+func (e *hooks) getPayload(ctx context.Context, cr *svcapitypes.SecretParameters) ([]byte, error) {
```

Comment:
`cr` seems to be used to point to the whole `Secret` object in this file, i.e. as in `(c)ustom (r)esource`. So, I'd suggest either keeping this line as is or use `params` as variable name when it addresses `SecretParameters`.

Reply:
Renamed to `params`.